File: Team_Workspace/dalia_ahmed/task8.py

class logger:
    def __init__(self):
      print("--- LOG STARTED ---")
    def __del__(self):
       print("--- LOG SAVED & CLOSED ---")

print ("step1")

def test_scope():
   mylogger=logger()
   print ("Step 2 (Inside Function)")
test_scope()
print ("Step 3 (Outside Function)")

File: Team_Workspace/Fatma_Ahmed/cpp_tasks/task8/task8.cpp
// task8.

#include <iostream>
using namespace std;

class Logger {
public:
    Logger() {
        cout << "---- LOG STARTED ---" << endl;

    }
    ~Logger() {
        cout << "--- LOG SAVED & CLOSED ---" << endl;

    }
};

int main()
{
    cout << "step 1 \n";
    {
        Logger obj1;
        cout << "Step 2 (inside Scope)" << endl;
    }
    cout << "Steo 3 (outside Scope)" << endl;
}

